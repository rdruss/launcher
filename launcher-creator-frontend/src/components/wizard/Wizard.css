.wizard{padding-bottom:50px}.wizard ul{margin:0;padding:0;list-style:none}.wizard ul>li{display:block;position:relative}.wizard ul>li>a{display:block;padding:10px 10px 10px 40px;color:#333c4e;font-size:17px;font-weight:400;letter-spacing:.8px}.wizard ul>li>a:before{content:'';position:absolute;width:1px;height:calc(100% - 25px);background-color:#bdc2ce;left:13px;bottom:-9px;z-index:3}.wizard ul>li>a .ico{pointer-events:none;font-size:14px;position:absolute;left:10px;top:15px;z-index:2}.wizard ul>li>a:after{content:'';position:absolute;border:2px solid #bdc2ce;border-radius:50%;top:14px;left:6px;width:16px;height:16px;z-index:3}.wizard ul>li .content,.wizard ul>li .summary{display:block;padding:10px 10px 10px 40px}.wizard ul>li .content{color:#696e75}.wizard ul>li .summary{color:#5cb85c;font-size:11px;font-weight:400;letter-spacing:.8px}.wizard ul>li.completed>a:before{background-color:#5cb85c;opacity:1;height:calc(100% - 25px);bottom:-9px}.wizard ul>li.completed>a:after{display:none}.wizard ul>li.locked>a:after,.wizard ul>li:last-child>a:before{display:none}.wizard ul>li.completed>a .ico{left:8px}.wizard ul>li>a .ico.ico-green{color:#5cb85c}.wizard ul>li>a .ico.ico-muted{color:#bdc2ce}.wizard ul>li.selected{background-color:#fff}.wizard ul>li.selected>a:before{background-color:#ffe357;opacity:1}.wizard ul>li.selected>a:after{border-color:#ffe357;background-color:#ffe357;opacity:1}.wizard ul>li.selected:after,.wizard ul>li.selected:before{left:100%;top:50%;border:solid transparent;content:" ";height:0;width:0;position:absolute;pointer-events:none}.wizard ul>li.selected:after{border:10px rgba(255,255,255,0);border-left-color:#fff;margin-top:-10px}.wizard ul>li.selected:before{border:11px rgba(234,236,241,0);border-left-color:#eaecf1;margin-top:-11px}.wizard .wizard-step-footer{margin-top:30px}.wizard .wizard-step-footer .pf-c-button.pf-m-link{margin-left:20px}
