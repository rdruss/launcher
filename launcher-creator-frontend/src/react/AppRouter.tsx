import * as React from 'react';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import LoginPageContainer from './pages/LoginPageContainer';
import HomePageContainer from './pages/HomePageContainer';

class AppRouterProps {
  public readonly authenticated: boolean;
}

const AppRouter = ({ authenticated }: AppRouterProps) => {
  if (!authenticated) {
    return (
      <BrowserRouter>
        <Route path="/">
          <Switch>
            <Route path="/login" component={LoginPageContainer} />
            <Redirect from="/" to="/login" />
          </Switch>
        </Route>
      </BrowserRouter>
    );
  }
  return (
    <BrowserRouter>
      <Route path="/">
        <Switch>
          <Route exact={true} path="/" component={HomePageContainer} />
          <Redirect from="/" to="/" />
        </Switch>
      </Route>
    </BrowserRouter>
  );
};

export default AppRouter;