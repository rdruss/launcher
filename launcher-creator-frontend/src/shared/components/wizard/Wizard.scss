.wizard {
  padding-bottom: 50px;
  ul {
    margin: 0;
    padding: 0;
    list-style: none;
    > li {
      display: block;
      position: relative;
      > a {
        display: block;
        padding: 10px 10px 10px 40px;
        color: #333c4e;
        font-size: 17px;
        font-weight: 400;
        letter-spacing: .8px;
        &:before {
          content: '';
          position: absolute;
          width: 1px;
          height: calc(100% - 25px);
          background-color: #bdc2ce;
          left: 13px;
          bottom: -9px;
          z-index: 3;
        }
        .ico {
          pointer-events: none;
          font-size: 14px;
          position: absolute;
          left: 10px;
          top: 15px;
          z-index: 2;
        }
        &:after {
          content: '';
          position: absolute;
          border: 2px solid #bdc2ce;
          border-radius: 50%;
          top: 14px;
          left: 6px;
          width: 16px;
          height: 16px;
          z-index: 3;
        }
      }

      .content, .summary {
        display: block;
        padding: 10px 10px 10px 40px;
      }

      .content {
        color: #696e75;
      }

      .summary {
        color: #5cb85c;
        font-size: 11px;
        font-weight: 400;
        letter-spacing: .8px;
      }
      &.completed > a {
        &:before {
          background-color: #5cb85c;
          opacity: 1;
          height: calc(100% - 25px);
          bottom: -9px;
        }
        &:after {
          display: none;
        }
      }
      &.locked > a:after, &:last-child > a:before {
        display: none;
      }
      &.completed > a .ico {
        left: 8px;
      }
      > a .ico {
        &.ico-green {
          color: #5cb85c;
        }
        &.ico-muted {
          color: #bdc2ce;
        }
      }
      &.selected {
        background-color: #fff;
        > a {
          &:before {
            background-color: #ffe357;
            opacity: 1;
          }
          &:after {
            border-color: #ffe357;
            background-color: #ffe357;
            opacity: 1;
          }
        }
        &:after, &:before {
          left: 100%;
          top: 50%;
          border: solid transparent;
          content: " ";
          height: 0;
          width: 0;
          position: absolute;
          pointer-events: none;
        }
        &:after {
          border: 10px rgba(255, 255, 255, 0);
          border-left-color: #fff;
          margin-top: -10px;
        }
        &:before {
          border: 11px rgba(234, 236, 241, 0);
          border-left-color: #eaecf1;
          margin-top: -11px;
        }
      }
    }
  }

  .wizard-step-footer {
    margin-top: 30px;

    .pf-c-button.pf-m-link {
      margin-left: 20px;
    }

  }
}