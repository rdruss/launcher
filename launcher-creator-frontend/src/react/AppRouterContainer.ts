import { connect } from 'react-redux';
import { AppState } from '../redux/states';
import AppRouter from './AppRouter';

const mapStateToProps = (state: AppState) => ({
  authenticated: state.authentication.authenticated,
});

const mapDispatchToProps = (dispatch) => ({
});

const AppRouterContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppRouter);

export default AppRouterContainer;
