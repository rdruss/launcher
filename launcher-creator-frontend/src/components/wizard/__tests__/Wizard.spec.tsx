import * as React from 'react';
import { shallow } from 'enzyme';
import Wizard from '../index';

describe('<Wizard />', () => {
  it('renders the wizard with one step', () => {
    const component = shallow(<Wizard><Wizard.Step title={'my first step'} selected={true} locked={false}/></Wizard>);
    expect(component).toMatchSnapshot();
  });

});

