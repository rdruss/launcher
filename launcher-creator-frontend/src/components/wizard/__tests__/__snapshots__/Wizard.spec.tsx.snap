// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wizard /> renders the wizard with one step 1`] = `
<div
  className="wizard"
>
  <ul>
    <WizardStep
      completed={false}
      locked={false}
      selected={true}
      title="my first step"
    />
  </ul>
</div>
`;
